### PR TITLE
PARQUET-2251 Avoid generating Bloomfilter when all pages of a column are encoded by dictionary in parquet v1

### PR DESCRIPTION
**Pull request by yabola**
In parquet pageV1, even all pages of a column are encoded by dictionary, it will still generate BloomFilter. Actually it is unnecessary and it costs time and occupies storage. Parquet pageV2 doesn't have this problem.

### CODE REVIEW

**Comment by yabola** [comment]
@wgtmac @gerashegalov Please take a look, thank you~ 
And I will update  [PR](https://github.com/apache/parquet-mr/pull/1023) to skip bloomfilter when all pages are encoded in dictionary (because there are already many wrong parquet files generated)

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestStoreBloomFilter.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  public TestStoreBloomFilter(Path file, String pageVersion) {
+    this.file = file;
+    this.pageVersion = pageVersion;
+  }
+
+  @Parameterized.Parameters(name = "Run {index}: parquet {1}")
+  public static Collection<Object[]> params() {
+    return Arrays.asList(
+      new Object[]{FILE_V1, "pageV1"},
```
Comment (wgtmac) [inline parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestStoreBloomFilter.java:66]: nit: the name is a little bit confusing. file format version is not same as data page version.

**Review by wgtmac — APPROVED** [review]
LGTM.

cc @gszadovszky

File: parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestStoreBloomFilter.java (hunk 1 above)
Comment (yabola) [reply parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestStoreBloomFilter.java]: thanks, I modified the description

**Review by gszadovszky — APPROVED** [review]
Thanks, @yabola, @wgtmac.

**Comment by yabola** [comment]
@wgtmac @gszadovszky Thank you for your review and help

**Pull request merged**